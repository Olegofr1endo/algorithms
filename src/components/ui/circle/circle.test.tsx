import React from "react"
import { Circle } from "./circle"
import TestRenderer from "react-test-renderer";
import { ElementStates } from "../../../types/element-states";

const reactElement = <div>div-block</div>

describe("Circle component tests", ()=>{
    it("without letter", ()=>{
        const tree = TestRenderer.create(<Circle />).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it("with letter", ()=>{
        const tree = TestRenderer.create(<Circle letter="A"/>).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it("with head", ()=>{
        const tree = TestRenderer.create(<Circle head="head"/>).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it("with element-head", ()=>{
        const tree = TestRenderer.create(<Circle head={reactElement}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it("with tail", ()=>{
        const tree = TestRenderer.create(<Circle tail="tail"/>).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it("with element-tail", ()=>{
        const tree = TestRenderer.create(<Circle tailType="element" tail={reactElement}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it("with index", ()=>{
        const tree = TestRenderer.create(<Circle index={0}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it("small", ()=>{
        const tree = TestRenderer.create(<Circle isSmall={true}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it("default", ()=>{
        const tree = TestRenderer.create(<Circle state={ElementStates.Default}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it("changing", ()=>{
        const tree = TestRenderer.create(<Circle state={ElementStates.Changing}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it("modified", ()=>{
        const tree = TestRenderer.create(<Circle state={ElementStates.Modified}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
})