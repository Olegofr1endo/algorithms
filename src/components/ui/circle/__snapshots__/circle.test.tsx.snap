// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle component tests changing 1`] = `
<div
  className="content "
  data-cypress="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cypress="circle-head"
  />
  <div
    className="circle   changing"
    data-cypress="circle-color"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cypress="circle-index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cypress="circle-tail"
  />
</div>
`;

exports[`Circle component tests default 1`] = `
<div
  className="content "
  data-cypress="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cypress="circle-head"
  />
  <div
    className="circle   default"
    data-cypress="circle-color"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cypress="circle-index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cypress="circle-tail"
  />
</div>
`;

exports[`Circle component tests modified 1`] = `
<div
  className="content "
  data-cypress="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cypress="circle-head"
  />
  <div
    className="circle   modified"
    data-cypress="circle-color"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cypress="circle-index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cypress="circle-tail"
  />
</div>
`;

exports[`Circle component tests small 1`] = `
<div
  className="content "
  data-cypress="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cypress="circle-head"
  />
  <div
    className="circle  small default"
    data-cypress="circle-color"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cypress="circle-index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cypress="circle-tail"
  />
</div>
`;

exports[`Circle component tests with element-head 1`] = `
<div
  className="content "
  data-cypress="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cypress="circle-head"
  >
    <div>
      div-block
    </div>
  </div>
  <div
    className="circle   default"
    data-cypress="circle-color"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cypress="circle-index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cypress="circle-tail"
  />
</div>
`;

exports[`Circle component tests with element-tail 1`] = `
<div
  className="content "
  data-cypress="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cypress="circle-head"
  />
  <div
    className="circle   default"
    data-cypress="circle-color"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cypress="circle-index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cypress="circle-tail"
  >
    <div>
      div-block
    </div>
  </div>
</div>
`;

exports[`Circle component tests with head 1`] = `
<div
  className="content "
  data-cypress="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
    data-cypress="circle-head"
  >
    head
  </div>
  <div
    className="circle   default"
    data-cypress="circle-color"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cypress="circle-index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cypress="circle-tail"
  />
</div>
`;

exports[`Circle component tests with index 1`] = `
<div
  className="content "
  data-cypress="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cypress="circle-head"
  />
  <div
    className="circle   default"
    data-cypress="circle-color"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cypress="circle-index"
  >
    0
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
    data-cypress="circle-tail"
  />
</div>
`;

exports[`Circle component tests with letter 1`] = `
<div
  className="content "
  data-cypress="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cypress="circle-head"
  />
  <div
    className="circle   default"
    data-cypress="circle-color"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      A
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cypress="circle-index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cypress="circle-tail"
  />
</div>
`;

exports[`Circle component tests with tail 1`] = `
<div
  className="content "
  data-cypress="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cypress="circle-head"
  />
  <div
    className="circle   default"
    data-cypress="circle-color"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cypress="circle-index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
    data-cypress="circle-tail"
  >
    tail
  </div>
</div>
`;

exports[`Circle component tests without letter 1`] = `
<div
  className="content "
  data-cypress="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cypress="circle-head"
  />
  <div
    className="circle   default"
    data-cypress="circle-color"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cypress="circle-index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cypress="circle-tail"
  />
</div>
`;
